culinary-canvas/capstone-project/frontend/my_capstone_project/src/UserRecommendation.jsx

```diff
@@ -0,0 +1,110 @@
+import LoadingSpinner from './LoadingSpinner';
+
+function UserRecommendation(){
+    const navigate = useNavigate();
+    const {user, login} = useAuth();
+    const [cookingLevel, setCookingLevel] = useState('');
+    const [dietaryPreferences, setDietaryPreferences] = useState([]);
+    const [favoriteFoods, setFavoriteFoods] = useState('');
+    const [loading, setLoading] = useState(false)
+    const backendAddress =import.meta.env.VITE_BACKEND_ADDRESS
+
+    const handleCheckboxChange = (event) => {
+        const {value, checked} = event.target;
+        setDietaryPreferences((prev) => checked ? [...prev, value] : prev.filter((pref) => pref !== value))
+    };
+
+    const handleSubmit = async(event) => {
+        event.preventDefault();
+        setLoading(true)
+        if(user){
+            try{
+                const response =  await fetch(`${backendAddress}/save-preferences`, {
+                    method: 'POST',
+                    headers: {
+                        'Content-Type' : 'application/json'
+                    },
+                    body: JSON.stringify({
+                        userId: user.id,
+                        cookingLevel,
+                        dietaryPreferences,
+                        favoriteFoods: favoriteFoods.split(',').map(food => food.trim()).filter(food => food !== '')}),
+                    credentials: 'include',
+                });
+                const data = await response.json()
+                console.log(data)
+                if(response.ok){
+                    alert('Preferences saved successfully. Please log in now');
+                    navigate('/login');
+                }else{
+                    alert('Profile Set Up Failed')
+                }
+            }catch(error){
+                console.error('Error during profile setup: ', error)
+                alert('Profile Set Up Failed')
+            }finally{
+                setLoading(false)
+            }
+        }else{
+            alert('User is not authenticated');
+            setLoading(false)
+        }
+
+    }
+
+    return(
+        <form onSubmit={handleSubmit}>
+            <h2>Profile SetUp</h2>
+            <div>
+                <label>Cooking Level: </label>
+                <select value={cookingLevel} onChange={(e) => setCookingLevel(e.target.value)} required>
+                    <option value="">Select your cooking level</option>
+                    <option value="beginner">Beginner</option>
+                    <option value="intermediate">Intermediate</option>
+                    <option value="advanced learner">Advanced Learner</option>
+                    <option value="Professional">Professional</option>
+                </select>
+            </div>
+            <div>
+                <label>Dietary Preferences (Optional): </label>
+                <div>
+                    <label><input type='checkbox' value="Gluten-free" onChange={handleCheckboxChange}/>Gluten-free</label>
```

Comment:
Please change this. I don't think it's good to hardcode a long list of labels like this. They're all the same format, so this should be easy to abstract away.

What you should do is have some type of array that stores each dietary preference, e.g. `["Gluten-free", "Sugar-Conscious", ...]`.

Then, here, you should go through each element in that array and create a new label.